src/Compilers/CSharp/Portable/Binder/Binder_Symbols.cs

```diff
@@ -811,21 +811,37 @@ private static bool IsViableType(LookupResult result)
             this.LookupSymbolsSimpleName(result, qualifierOpt, identifierValueText, 0, basesBeingResolved, options, diagnose: true, useSiteDiagnostics: ref useSiteDiagnostics);
             diagnostics.Add(node, useSiteDiagnostics);
 
-            Symbol bindingResult;
-            // If we were looking up the identifier "dynamic" at the topmost level and didn't find anything good,
-            // we actually have the type dynamic (assuming /langversion is at least 4).
+            Symbol bindingResult = null;
+
+            // If we were looking up "dynamic" or "nint" at the topmost level and didn't find anything good,
+            // use that particular type (assuming the /langversion is supported).
             if ((object)qualifierOpt == null &&
                 (node.Parent == null ||
-                 node.Parent.Kind() != SyntaxKind.Attribute && // dynamic not allowed as an attribute type
+                 node.Parent.Kind() != SyntaxKind.Attribute && // dynamic and nint not allowed as attribute type
                  SyntaxFacts.IsInTypeOnlyContext(node)) &&
```

Comment:
>SyntaxFacts.IsInTypeOnlyContext(node) [](start = 17, length = 37)

Not sure this constraint `SyntaxFacts.IsInTypeOnlyContext(node)` makes sense for the native types.  For example, `nint.Zero` is a perfectly valid access to a static field.

Reply:
The following works today:
```csharp
_ = int.Equals(0, 0);
```
Please add a corresponding test using `nint` instead of `int`.

---
In reply to: [334613881](https://github.com/dotnet/roslyn/pull/39157#discussion_r334613881) [](ancestors = 334613881)

Reply:
Updated `NativeIntTests.MemberName()` to use `nint.Equals(,)`. I don't believe that particular form uses this binding method however.

---
In reply to: [354534988](https://github.com/dotnet/roslyn/pull/39157#discussion_r354534988) [](ancestors = 354534988,334613881)